contrib/babelfishpg_tds/src/backend/tds/tdsprotocol.c

```diff
@@ -398,6 +408,10 @@ ProcessTDSRequest(TDSRequest request)
 				Assert(0);
 				break;
 		}
+
+		/* Disabling statement timeout after processing. */
+		disable_statement_timeout();
```

Comment:
how are we making sure that a running timer is not leaked due error conditions etc?

Reply:
We should call this function as the first statement in PG_CATCH block as well.